llpc/context/llpcCompiler.h

```diff
@@ -69,23 +70,15 @@ class GraphicsShaderCacheChecker {
   // Check shader caches, returning mask of which shader stages we want to keep in this compile.
   unsigned check(const llvm::Module *module, unsigned stageMask, llvm::ArrayRef<llvm::ArrayRef<uint8_t>> stageHashes);
 
-  // Get cache results.
-  ShaderEntryState getNonFragmentCacheEntryState() { return m_nonFragmentCacheEntryState; }
-  ShaderEntryState getFragmentCacheEntryState() { return m_fragmentCacheEntryState; }
-
   // Update shader caches with results of compile, and merge ELF outputs if necessary.
   void updateAndMerge(Result result, ElfPackage *pipelineElf);
   void updateRootUserDateOffset(ElfPackage *pipelineElf);
 
 private:
   Compiler *m_compiler;
   Context *m_context;
-
-  // Old shader cache
-  ShaderEntryState m_nonFragmentCacheEntryState = ShaderEntryState::New;
-  ShaderCache *m_nonFragmentShaderCache = nullptr;
-  CacheEntryHandle m_hNonFragmentEntry = {};
-  BinaryData m_nonFragmentElf = {};
+  std::unique_ptr<CacheAccessor> m_nonFragmentCacheAccessor;
+  std::unique_ptr<CacheAccessor> m_fragmentCacheAccessor;
```

Comment:
Since there's only one concrete that is used here (i.e., no subclasses), consider using `llvm::Optional<CacheAccessor>` to avoid unnecessary heap allocations.

Reply:
Same in the other PR